src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizSubmissionResource.java

```diff
@@ -95,11 +95,18 @@ public QuizSubmissionResource(QuizSubmissionRepository quizSubmissionRepository,
                         public void run() {
                             Participation participation = participationService.findOneByExerciseIdAndStudentLoginAnyState(exerciseId, principal.getName());
                             submitSubmission(participation, null);
+                            // notify user about new result
+                            messagingTemplate.convertAndSend("/topic/participation/" + participation.getId() + "/newResults", true);
                         }
-                    }, ZonedDateTime.now().until(quizExercise.getDueDate().plusSeconds(3), ChronoUnit.MILLIS));
+                    }, (quizExercise.getRemainingTime() + 3) * 1000);
```

Comment:
Can you create a constant for 3 seconds and 2 seconds (respectively -2 seconds)? Best would be in one of the constants or properties files